### PR TITLE
Updates1/doc/pytorch lightning

### PR DESCRIPTION
**Pull request by zredeaux65**
## What does this PR do?
This PR provides updates to the following RST files: data, speed, converting, installation, and style guide.

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/starter/style_guide.rst [hunk 1]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
+* What transforms were used?
 
-********
-Examples
-********
+It's for this reason that we recommend you use datamodules. This is especially important when collaborating because
+it will save your team a lot of time as well.
 
-Checkout the live examples to get your hands dirty:
+All they need to do is drop a datamodule into a lightning trainer and not worry about what was done to the data.
```
Comment (awaelchli) [inline docs/source/starter/style_guide.rst:204]:
```suggestion
All they need to do is drop a datamodule into a Lightning trainer and not worry about what was done to the data.
```
the brand name should be capitalized, right?

File: docs/source/guides/data.rst [hunk 2]
```diff
@@ -47,23 +47,21 @@ Read :ref:`this <datamodules>` for more details on LightningDataModule.
...
-   will automatically combine the batches from different DataLoaders. You can control the way how dataloaders of different length
-   are combined by the flag `multiple_trainloader_mode` of the :class:`~pytorch_lightning.trainer.Trainer`. Alternatively, you can provide
-   dataloaders via :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`.
-3. In the validation, test or prediction you have the option to either return multiple DataLoaders as list/tuple, which Lightning will call sequentially,
-   or combine the dataloaders using :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`, which Lightning will
+1. Create a DataLoader that iterates over multiple datasets under the hood.
+2. In the training loop, you can pass multiple DataLoaders as a dict or list/tuple, and Lightning will
+   automatically combine the batches from different DataLoaders.
+3. In the validation, test or prediction, you have the option to return multiple DataLoaders as list/tuple, which Lightning will call sequentially
+   or combine the DataLoaders  using :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`, which Lightning will
```
Comment (tchaton) [inline docs/source/guides/data.rst:56]:
```suggestion
   or combine the DataLoaders using :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`, which Lightning will
```

File: docs/source/guides/speed.rst [hunk 3]
```diff
@@ -251,7 +251,7 @@ less memory bandwidth and run match operations much faster on GPUs that support
 
 |
 
-Mixed precision combines the use of both 32 and 16 bit floating points to reduce memory footprint during model training, resulting in improved performance, achieving +3X speedups on modern GPUs.
+Mixed precision combines the use of both 32 and 16-bit floating points to reduce memory footprint during model training, resulting in improved performance, achieving +3X speedups on modern GPUs.
```
Comment (tchaton) [inline docs/source/guides/speed.rst:254]:
```suggestion
Mixed precision combines the use of both 32 and 16-bit floating points to reduce memory footprint during model training, resulting in improved performance, achieving up to +3X speedups on modern GPUs.
```

File: docs/source/starter/style_guide.rst [hunk 4]
```diff
@@ -26,58 +26,40 @@ In Lightning we differentiate between a system and a model.
...
+        def __init__(self, encoder: nn.Module = None, decoder: nn.Module = None):
             super().__init__()
-            self.layer_1 = nn.Linear()
-            self.layer_2 = nn.Linear()
-            self.layer_3 = nn.Linear()
+            self.encoder = encoder
+            self.decoder = decoder
 
+For fast prototyping, it's often useful to define all the computations in a LightningModule. For reusability
+and scalability, it might be better to pass in the relevant backbones.
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:62]:
```suggestion
and scalability, it might be better to pass in the relevant backbones or even better, to pass a single nn.Module.
```

File: docs/source/starter/style_guide.rst [hunk 5]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
 
 Forward vs training_step
 ========================
-
-We recommend using forward for inference/predictions and keeping ``training_step`` independent.
+We recommend using forward for inference/predictions and keeping training_step independent
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:154]: I prefer when training_step stands out. Might be worth using LightningModule.training_step method hyperlink.

File: docs/source/starter/style_guide.rst [hunk 6]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
 .. code-block:: python
 
     def forward(self, x):
         embeddings = self.encoder(x)
-        return embeddings
 
 
-    def training_step(self, batch, batch_idx):
-        x, y = batch
+    def training_step(self):
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:162]:
```suggestion
    def training_step(self, batch, batch_idx):
```

This is required and would break otherwise.

File: docs/source/starter/style_guide.rst [hunk 7]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
 Data
 ****
-
 These are best practices for handling data.
 
 Dataloaders
 ===========
-
-Lightning uses :class:`~torch.utils.data.DataLoader` to handle all the data flow through the system. Whenever you structure dataloaders,
+Lightning uses dataloaders to handle all the data flow through the system. Whenever you structure dataloaders,
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:187]: Let's keep :class:`~torch.utils.data.DataLoader`. The more connection to PyTorch Objects, the more transparent it is for the users.

File: docs/source/starter/style_guide.rst [hunk 8]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
 make sure to tune the number of workers for maximum efficiency.
 
-.. warning:: Make sure not to use ``Trainer(strategy="ddp_spawn")`` with ``num_workers>0`` in a DataLoader or you will bottleneck your code.
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:207]: Let's revert. Showing how to use the Trainer is key here.

File: docs/source/starter/style_guide.rst [hunk 9]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
+.. warning:: Make sure not to use ddp_spawn with num_workers > 0 or you will bottleneck your code.
 
 DataModules
 ===========
+Lightning introduced datamodules. The problem with dataloaders is that sharing full datasets is often still challenging
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:194]: I think this can be re-formulated.

File: docs/source/starter/style_guide.rst [hunk 10]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
+because all these questions need to be answered:
 
-The :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is designed as a way of decoupling data-related
-hooks from the :class:`~pytorch_lightning.core.lightning.LightningModule` so you can develop dataset agnostic models. It makes it easy to hot swap different
-datasets with your model, so you can test it and benchmark it across domains. It also makes sharing and reusing the exact data splits and transforms across projects possible.
-
-Check out :ref:`data` document to understand data management within Lightning and its best practices.
-
-------------
+* What splits were used?
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:197]:
```suggestion
* What dataset splits were used?
```

File: docs/source/starter/style_guide.rst [hunk 11]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
+* How many samples does this dataset have?
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:198]:
```suggestion
* How many samples does this dataset have overall and within each split?
```

File: docs/source/starter/style_guide.rst [hunk 12]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
+* What transforms were used?
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:199]:
```suggestion
* Which transforms were used?
```

File: docs/source/starter/style_guide.rst [hunk 13]
```diff
@@ -173,55 +151,57 @@ In practice, this code looks like:
...
-********
+It's for this reason that we recommend you use datamodules. This is especially important when collaborating because
+it will save your team a lot of time as well.
 
-Checkout the live examples to get your hands dirty:
+All they need to do is drop a datamodule into a lightning trainer and not worry about what was done to the data.
 
-- `Introduction to PyTorch Lightning <https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/mnist-hello-world.html>`_
-- `Introduction to DataModules <https://pytorch-lightning.readthedocs.io/en/stable/notebooks/lightning_examples/datamodules.html>`_
+This is true for both academic and corporate settings where data cleaning and ad-hoc instructions slow down the progress
```
Comment (tchaton) [inline docs/source/starter/style_guide.rst:206]: This section is a good start but I would add more content to it. Also, let's keep hyperlinks.

File: docs/source/guides/data.rst (hunk 2 above)
Comment (rohitgr7) [inline docs/source/guides/data.rst:56]:
```suggestion
3. In the validation, test, or prediction, you have the option to return multiple DataLoaders as list/tuple, which Lightning will call sequentially
or combine the DataLoaders using :class:`~pytorch_lightning.trainer.supporters.CombinedLoader`, which Lightning will
```

File: docs/source/guides/data.rst [hunk 14]
```diff
@@ -47,23 +47,21 @@ Read :ref:`this <datamodules>` for more details on LightningDataModule.
...
    automatically combine the batches from different DataLoaders.
 
 
 Using LightningDataModule
 =========================
 
-You can set more than one :class:`~torch.utils.data.DataLoader` in your :class:`~pytorch_lightning.core.datamodule.LightningDataModule` using its dataloader hooks
-and Lightning will use the correct one under-the-hood.
+You can set more than one :class:`~torch.utils.data.DataLoader` in your :class:`~pytorch_lightning.core.datamodule.LightningDataModule` using its DataLoader hooks
```
Comment (rohitgr7) [inline docs/source/guides/data.rst:63]:
```suggestion
You can set more than one :class:`~torch.utils.data.DataLoader` in your :class:`~pytorch_lightning.core.datamodule.LightningDataModule` using its dataLoader hooks
```
it's `val_dataloader`, `test_dataloader`, so on....

File: docs/source/guides/data.rst [hunk 15]
```diff
@@ -275,7 +254,7 @@ set is not available at the time your model was declared. Simply pass the test s
 Accessing DataLoaders within LightningModule
 ********************************************
 
-In the case that you require access to the DataLoader or Dataset objects, DataLoaders for each step can be accessed using the ``Trainer`` object:
+In the case that you require access to the DataLoader or dataset objects, DataLoaders for each step can be accessed using the ``Trainer`` object:
```
Comment (rohitgr7) [inline docs/source/guides/data.rst:257]: same here: `Dataset`

File: docs/source/guides/data.rst [hunk 16]
```diff
@@ -291,7 +270,7 @@ In the case that you require access to the DataLoader or Dataset objects, DataLo
             # extract metadata, etc. from the dataset:
             ...
 
-If you are using a :class:`~pytorch_lightning.trainer.supporters.CombinedLoader` object which allows you to fetch batches from a collection of dataloaders dataloader simultaneously which supports collections of dataloaders such as list, tuple, or dictionary. The dataloaders can be accessed using the same collection structure:
+If you are using a :class:`~pytorch_lightning.trainer.supporters.CombinedLoader` object which allows you to fetch batches from a collection of DataLoaders DataLoader simultaneously which supports collections of DataLoader such as list, tuple, or dictionary. The DataLoaders can be accessed using the same collection structure:
```
Comment (rohitgr7) [inline docs/source/guides/data.rst:273]:
```suggestion
If you are using a :class:`~pytorch_lightning.trainer.supporters.CombinedLoader` object which allows you to fetch batches from a collection of DataLoaders simultaneously which supports collections of DataLoader such as list, tuple, or dictionary. The DataLoaders can be accessed using the same collection structure:
```

File: docs/source/guides/data.rst (hunk 14 above)
Comment (Borda) [reply docs/source/guides/data.rst:63]: why capital L?

Comment (rohitgr7) [reply docs/source/guides/data.rst:63]: updated.. I just updated `D` before :)

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
